src/azure/cli/commands/__init__.py

```diff
@@ -142,8 +142,9 @@ def __init__(self, name, handler, description=None, table_transformer=None):
         self.table_transformer = table_transformer
 
     def add_argument(self, param_name, *option_strings, **kwargs):
+        dest = kwargs.pop('dest', None)
         argument = CliCommandArgument(
-            param_name, options_list=option_strings, **kwargs)
+            dest or param_name, options_list=option_strings, **kwargs)
```

Comment:
'dest' is a parameter supported by argparse. If you use 'argparse.SUPPRESS' for dest, then you can have the option in your command signature but NOT have it become part of the namespace, which obviates the need to later suppress the parameter. The only downside is that it means it doesn't end up in our command table... so that's why this little fix was necessary--to get the best of both worlds :)

---
In reply to: [77102829](https://github.com/Azure/azure-cli/pull/803#discussion_r77102829) [](ancestors = 77102829)